### ISSUE TITLE
`npm config edit --global` errors when `globalconfig` path does not exist

### ISSUE THREAD

**Issue by smikes** [npm/npm#7233]
The error message should convey more information about what the problem is.

Ref case #5881, a slightly different problem, but the one where I noticed this.

```
$ npm config edit --global
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/Users/smikes/.nvm/versions/io.js/v1.0.4/bin/npm" "config" "edit" "--global"
npm ERR! node v1.0.4
npm ERR! npm  v2.3.0
npm ERR! path /Users/smikes/.nvm/versions/io.js/v1.0.4/etc/npmrc.c33db61dc6e13a89a2ab11f8f1793b9e
npm ERR! code ENOENT
npm ERR! errno -2

npm ERR! enoent ENOENT, open '/Users/smikes/.nvm/versions/io.js/v1.0.4/etc/npmrc.c33db61dc6e13a89a2ab11f8f1793b9e'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/smikes/src/github/npm-kludge-search/npm-debug.log
```

**Comment by ljharb** [comment]
It seems like `npm config edit` should use something like `mkdir -p` to create the directory path to the config file?

**Comment by othiym23** [comment]
Yes, it probably should, and it probably doesn't for the simple reason that it's really easy to forget that `--globalconfig` is a thing. A patch fixing that would be most welcome, and probably pretty easy.

**Comment by othiym23** [comment]
Closed by #7249. :fireworks:
